### DISCUSSION TITLE
ordering

### DISCUSSION THREAD

**Discussion by keizah7** [spatie/laravel-translatable#363] [Q&A]
How I can order by translatable column using eloquent without harcoding locale?

```php
$locale = 'en';

$query->orderBy('attribute_title->' . $locale);
```

**Comment by ghost** [comment]
From the Laravel docs https://laravel.com/docs/9.x/localization#determining-the-current-locale

`
use Illuminate\Support\Facades\App;
 
$locale = App::currentLocale();
`

**Reply by keizah7** [reply]
Thanks, I know how get locale, I'm searching for way to order without providing it.

**Comment by michaelravedoni** [comment]
Hello @keizah7,
I suggest to use the Laravel collection method [sortBy()](https://laravel.com/docs/9.x/collections#method-sortby). For example :

```php
$queryData = $query->get(); // or Model::get();
$collectionSorted = $queryData->sortBy(function($item, $key){
    return $item->attribute_title;
});
$sorted = $collectionSorted->values();
```
I hope it can help you.

**Reply by keizah7** [reply]
Hello, I am using this query in Laravel Nova `indexQuery`, as you maybe know, that method should return query, so collection helpers is not usable in this case

**Reply by michaelravedoni** [reply]
Hello, oh okay. I can't help you more then. I have no experience with Laravel Nova. But since a `query` refers to columns, maybe it is possible to do something with a [`virtualAs()`](https://laravel.com/docs/9.x/migrations#column-modifiers) column in the migration file?
